src/vehicle.cpp

```diff
@@ -225,6 +225,18 @@ vehicle::~vehicle()
 {
 }
 
+void vehicle::set_hp( vehicle_part &pt, int qty )
+{
+    double k = pt.base.max_damage() / double( pt.info().durability );
+    pt.base.set_damage( pt.base.max_damage() - ( qty * k ) );
+}
+
+bool vehicle::mod_hp( vehicle_part &pt, int qty, damage_type dt )
+{
+    double k = pt.base.max_damage() / double( pt.info().durability );
+    return pt.base.mod_damage( - qty * k, dt );
```

Comment:
> Then it's a tangled logic that doesn't seem to be consistent with anything.

Consistent with `set_damage` and `mod_damage`

> At the very least you could extract this part into a separate function:

No. The two functions are adjacent and are both only two lines long. There's nothing to be gained in terms of clarity by factoring out a third single line function.

> Also, if pt.info().durability is zero

Never occurs and specifically excluded via a JSON check at `veh_type.cpp:491`